javascript/organizing_your_javascript_code/objects_and_object_constructors.md

```diff
@@ -149,94 +149,204 @@ console.log(theHobbit.info());
+In some places you might also come across `[[Prototype]]`, which is just another way of talking about the `.__proto__` property of an object, like `player1.[[Prototype]]`.
+
+This explanation about the `prototype` might have been a lot, so remember to take a breather before moving on!
+
+#### Prototypal Inheritance
+
+Now, you may also have a question - what's with the `__proto__` / `[[Prototype]]` of an object pointing to its `prototype`? What is the purpose of defining properties and functions on the `prototype`?
+
+We can narrow it down to two reasons:
+
+1. We can define properties and functions common among all objects on the `prototype` to save memory. Defining every property and function takes up a lot of memory, especially if you have a lot of common properties and functions, and a lot of created objects! Defining them on a centralized, shared object which the objects have access to, thus saves memory.
+
+2. The second reason is the name of this section, **Prototypal Inheritance**, which we've referred to in passing earlier, in the introduction to the Prototype. In recap, we know that can say that the `player1` or `player2` objects _inherit_ from the `Player.prototype` object, like with the `.sayHello` function, by being able to access it.
+
+Let's now try to do the following:
+
+~~~javascript
+// Player.prototype.__proto__ 
+Object.getPrototypeOf(Player.prototype) === Object.prototype // true
+
+// Output may slightly differ based on the browser
+player1.valueOf() // Output: Object { name: "steve", marker: "X", sayName: sayName() } 
+~~~
+
+What's this `.valueOf` function, and where did it come from if we did not define it? It comes as a result of `Player.prototype.__proto__` having the value of `Object.prototype`! This means that `Player.prototype` is inheriting from `Object.prototype`. This `.valueOf` function is defined on `Object.prototype` just like `.sayHello` is defined on `Player.prototype`.
+
+Essentially, this is how JavaScript makes use of `prototype` and `__proto__` - by having the `__proto__` values of objects point to `prototype`s and inheriting from those prototypes, and thus forming a chain. This kind of inheritance using prototypes is hence named as Prototypal inheritance. JavaScript figures out which properties exist (or do not exist) on the object and starts traversing the chain to find the property or function, like so:
+
+1. Is the `.valueOf` function part of the `player1` object? No, it is not. (Remember, only the `name`, `marker` and `sayName` properties are part of the `Player` objects.)
+
+2. Is the function part of the `player1`'s prototype (the `player1.__proto__` value, i.e., `Player.prototype`)? No, only the `.sayHello` function is a part of it.
+
+3. Well, then, is it part of `Player.prototype.__proto__` (=== `Object.prototype`)? Yes, `.valueOf` is defined on `Object.prototype`!
```

Comment:
Can we use this aside to also cover how `hasOwnProperty` works as well? It's in a very similar vein and is a method that is likely to be used or encountered.